smalltalksrc/VMMaker/DruidJIT.class.st

```diff
@@ -5046,35 +5042,35 @@ DruidJIT >> gen_PrimitiveNewWithArg [
 	self
 		DivR: ClassReg
 		R: Extra0Reg
-		Quo: ClassReg
-		Rem: Extra0Reg.
+		Quo: Extra0Reg
+		Rem: ClassReg.
 	s90 := 4.
-	self MoveCq: s90 R: Extra0Reg.
-	self SubR: SendNumArgsReg R: Extra0Reg.
-	self AndCq: 3 R: Extra0Reg.
-	self AddR: Extra0Reg R: Extra1Reg.
-	self MoveR: ClassReg R: SendNumArgsReg.
+	self MoveCq: s90 R: ClassReg.
+	self SubR: SendNumArgsReg R: ClassReg.
+	self AndCq: 3 R: ClassReg.
+	self AddR: ClassReg R: Extra1Reg.
+	self MoveR: Extra0Reg R: SendNumArgsReg.
 	self MoveCq: s14 R: Extra0Reg.
 	jump8 := self Jump: 0.
 	currentBlock := self Label.
 	jump3 jmpTarget: currentBlock.
 	self CmpCq: 16 R: Extra1Reg.
 	jump3 := self JumpNonZero: 0.
-	self MoveR: SendNumArgsReg R: Extra0Reg.
-	self AddCq: 7 R: Extra0Reg.
+	self MoveR: SendNumArgsReg R: ClassReg.
+	self AddCq: 7 R: ClassReg.
 	s102 := 8.
-	self MoveCq: s102 R: ClassReg.
+	self MoveCq: s102 R: Extra0Reg.
 	self
-		DivR: ClassReg
-		R: Extra0Reg
-		Quo: ClassReg
-		Rem: Extra0Reg.
+		DivR: Extra0Reg
+		R: ClassReg
+		Quo: Extra0Reg
+		Rem: ClassReg.
 	s105 := 8.
-	self MoveCq: s105 R: Extra0Reg.
-	self SubR: SendNumArgsReg R: Extra0Reg.
-	self AndCq: 7 R: Extra0Reg.
-	self AddR: Extra0Reg R: Extra1Reg.
-	self MoveR: ClassReg R: SendNumArgsReg.
+	self MoveCq: s105 R: ClassReg.
+	self SubR: SendNumArgsReg R: ClassReg.
```

Comment:
@PalumboN for Juan Ignacio. Could this be collapsed in a single instruction?

Reply:
https://github.com/Alamvic/druid/issues/106